app/assets/javascripts/discourse/app/components/topic-list-item.js

```diff
@@ -37,21 +37,16 @@ export function showEntrance(e) {
 }
 
 export function navigateToTopic(topic, href) {
-  const owner = getOwner(this);
-  const router = owner.lookup("service:router");
-  const session = owner.lookup("service:session");
-
-  session.set("lastTopicIdViewed", {
-    topicId: topic.id,
-    historyUuid: router.location.getState?.().uuid,
-  });
+  const historyStore = getOwner(this).lookup("service:history-store");
```

Comment:
Shouldn't this be `getOwnerWithFallback`?

Reply:
It was using `getOwner` previously with no issues. This function is a bit weird - it is never called directly. Instead, it is re-used in multiple classes. So when it's called, it's always in the context of an Ember component.